### ISSUE TITLE
Window loses focus when moving between workspaces for "Workspaces Span Displays"

### ISSUE THREAD

**Issue by jasonrhansen** [pop-os/cosmic-comp#865]
If the Multi-monitor Behavior is set to "Workspaces Span Displays" in the Workspaces settings, I experience the following issue. 

When moving windows between workspaces with keyboard shortcuts, the active/focused window loses it's focus and sometimes ends up below other windows.

Before moving the focused terminal window, on workspace 1.
![before-move](https://github.com/user-attachments/assets/6981b7f6-af4b-4cef-af53-114ece7fe68b)

After moving the terminal window to workspace 2 with Super+Ctrl+Shift+J. Notice how the App Store window is in front and focused instead of the terminal window that was just moved.
![after-move](https://github.com/user-attachments/assets/a3a58611-d838-4445-817f-0da1ac0c755a)

**Comment by jasonrhansen** [comment]
The same happens when moving windows between displays with the keyboard.

**Comment by WatchMkr** [comment]
Is the focus when moving workspaces issue fixed for you?

**Comment by jasonrhansen** [comment]
It's definitely still a problem for me with all packages up to date. When I originally reported this I was using NVIDIA graphics, so I was thinking it could possibly be a driver bug. But now I'm using a new Pangolin laptop with AMD graphics.

![Image](https://github.com/user-attachments/assets/960826fd-29d5-4c42-876d-dd37d51fd912)

One simple way to reproduce this issue is to focus a window on worspace 1, then use Super+Shift+2 to move it to workspace 2. Then immediately try moving it back to workspace 1 with Super+Shift+1. It doesn't work. I have to click on the window first to be able to move it again.

The second part of this issue is that if I move a maximized window to another workspace, it is no longer maximized, but I would like it to retain the maximized state. Maybe this should be split off into a separate issue?

I wonder if the window losing focus is related to #453, which is another focus issue that I've noticed.

**Comment by curiouslad** [comment]
> The second part of this issue is that if I move a maximized window to another workspace, it is no longer maximized, but I would like it to retain the maximized state. Maybe this should be split off into a separate issue?
> 
I agree that that window should stay maximized after moving, also i think that if there is only one window on tiled workspace it should be maximized. I often find my self needing to super+m to remove active window hint when there is only one window

**Comment by jokeyrhyme** [comment]
I wonder if this has been fixed in a git commit that has not yet been included in an alpha release?

**Comment by curiouslad** [comment]
> I wonder if this has been fixed in a git commit that has not yet been included in an alpha release?

it is possible, i just tried to reproduce on fedora and i couldnt.
(fedora package is usually close to master)

**Comment by jasonrhansen** [comment]
Unfortunately that's not it. I'm already on the latest commit on master.

```
cosmic-comp/noble,now 0.1~1741971214~24.04~7340e2b amd64 [installed,automatic]
```

I've also tested it in both floating and tiling modes and I have the same issue either way.

**Comment by jasonrhansen** [comment]
@WatchMkr I think I figured out why this issue isn't affecting you and others. It's a difference in the multi-monitor behavior in the workspace settings. If I have it set "Workspaces Span Displays", I have the issue. If it's set to "Displays Have Separate Workspaces", the issue goes away.

**Comment by awilkins** [comment]
See also #1365 which details a behaviour difference between "Workspace spans displays" and "Displays have separate workspaces).

**Comment by WatchMkr** [comment]
Tracking here: https://github.com/pop-os/cosmic-workspaces-epoch/issues/95

Some improvements have been made and well be testing internally and reporting results on that issue.
